File: app/src/main/java/com/blog/demo/People.java
package com.blog.demo;

public class People {
    public int id;
    public String name;
    public String addr;
    public int age;

    public People() {
    }

    public People(String name, String addr) {
        this.name = name;
        this.addr = addr;
    }

    public People(int id, String name, String addr, int age) {
        this.id = id;
        this.name = name;
        this.addr = addr;
        this.age = age;
    }

    public String description() {
        return id + ":" + name + ":" + addr + ":" + age;
    }
}

File: app/src/main/java/com/blog/demo/People.kt
package com.blog.demo

class People constructor(var id: Int, var name: String? = null, var addr: String?, var age: Int) {

    constructor() : this(null, null)

    constructor(name: String?, addr: String?) : this(0, name, addr, 0)

    fun description(): String {
        return  "$id:$name:$addr:$age"
    }

}